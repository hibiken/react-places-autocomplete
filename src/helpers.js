export const debounce = (func, wait, immediate) => {
  let timeout
  return () => {
    const args = arguments
    const later = () => {
      timeout = null
      if (!immediate) func(args)
    }
    const callNow = immediate && !timeout
    clearTimeout(timeout)
    timeout = setTimeout(later, wait)
    if (callNow) func(args)
  }
}
