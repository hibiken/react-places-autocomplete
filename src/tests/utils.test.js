import {
  geocodeByAddress,
  geocodeByPlaceId,
  geocodeByLocation,
} from '../utils';
import { setupGoogleMock } from './helpers/setup';

beforeAll(() => {
  setupGoogleMock();
});

describe('geocodeByAddress', () => {
  it('geocodes valid address', () => {
    expect.assertions(1);
    return geocodeByAddress('San Francisco, CA').then(results => {
      expect(results).toMatchSnapshot();
    });
  });

  it('rejects invalid address', () => {
    expect.assertions(1);
    return geocodeByAddress('someinvalidaddress').catch(error => {
      expect(error).toBeDefined();
    });
  });
});

describe('geocodeByPlaceId', () => {
  it('geocode valid placeID', () => {
    expect.assertions(1);
    /* placeID of San Francisco */
    return geocodeByPlaceId('ChIJIQBpAG2ahYAR_6128GcTUEo').then(results => {
      expect(results).toMatchSnapshot();
    });
  });
});

describe('geocodeByLocation', () => {
  it('geocode valid location', () => {
    expect.assertions(1);
    /* location of San Francisco */
    return geocodeByLocation({
      lat: 37.7749295,
      lng: -122.41941550000001,
    }).then(results => {
      expect(results).toMatchSnapshot();
    });
  });
});

/* TODO: test getLatLng */
