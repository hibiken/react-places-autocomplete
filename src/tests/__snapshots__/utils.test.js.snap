// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`geocodeByAddress geocodes valid address 1`] = `
Array [
  Object {
    "address_components": Array [
      Object {
        "long_name": "San Francisco",
        "short_name": "SF",
        "types": Array [
          "locality",
          "political",
        ],
      },
      Object {
        "long_name": "San Francisco County",
        "short_name": "San Francisco County",
        "types": Array [
          "administrative_area_level_2",
          "political",
        ],
      },
      Object {
        "long_name": "California",
        "short_name": "CA",
        "types": Array [
          "administrative_area_level_1",
          "political",
        ],
      },
      Object {
        "long_name": "United States",
        "short_name": "US",
        "types": Array [
          "country",
          "political",
        ],
      },
    ],
    "formatted_address": "San Francisco, CA, USA",
    "geometry": Object {
      "bounds": Object {
        "east": -122.28178000000003,
        "north": 37.9298239,
        "south": 37.6398299,
        "west": -123.17382499999997,
      },
      "location": Object {
        "lat": 37.7749295,
        "lng": -122.41941550000001,
      },
      "location_type": "APPROXIMATE",
      "viewport": Object {
        "east": -122.34820000000002,
        "north": 37.812,
        "south": 37.70339999999999,
        "west": -122.52699999999999,
      },
    },
    "place_id": "ChIJIQBpAG2ahYAR_6128GcTUEo",
    "types": Array [
      "locality",
      "political",
    ],
  },
]
`;

exports[`geocodeByLocation geocode valid location 1`] = `
Array [
  Object {
    "address_components": Array [
      Object {
        "long_name": "San Francisco",
        "short_name": "SF",
        "types": Array [
          "locality",
          "political",
        ],
      },
      Object {
        "long_name": "San Francisco County",
        "short_name": "San Francisco County",
        "types": Array [
          "administrative_area_level_2",
          "political",
        ],
      },
      Object {
        "long_name": "California",
        "short_name": "CA",
        "types": Array [
          "administrative_area_level_1",
          "political",
        ],
      },
      Object {
        "long_name": "United States",
        "short_name": "US",
        "types": Array [
          "country",
          "political",
        ],
      },
    ],
    "formatted_address": "San Francisco, CA, USA",
    "geometry": Object {
      "bounds": Object {
        "east": -122.28178000000003,
        "north": 37.9298239,
        "south": 37.6398299,
        "west": -123.17382499999997,
      },
      "location": Object {
        "lat": 37.7749295,
        "lng": -122.41941550000001,
      },
      "location_type": "APPROXIMATE",
      "viewport": Object {
        "east": -122.34820000000002,
        "north": 37.812,
        "south": 37.70339999999999,
        "west": -122.52699999999999,
      },
    },
    "place_id": "ChIJIQBpAG2ahYAR_6128GcTUEo",
    "types": Array [
      "locality",
      "political",
    ],
  },
]
`;

exports[`geocodeByPlaceId geocode valid placeID 1`] = `
Array [
  Object {
    "address_components": Array [
      Object {
        "long_name": "San Francisco",
        "short_name": "SF",
        "types": Array [
          "locality",
          "political",
        ],
      },
      Object {
        "long_name": "San Francisco County",
        "short_name": "San Francisco County",
        "types": Array [
          "administrative_area_level_2",
          "political",
        ],
      },
      Object {
        "long_name": "California",
        "short_name": "CA",
        "types": Array [
          "administrative_area_level_1",
          "political",
        ],
      },
      Object {
        "long_name": "United States",
        "short_name": "US",
        "types": Array [
          "country",
          "political",
        ],
      },
    ],
    "formatted_address": "San Francisco, CA, USA",
    "geometry": Object {
      "bounds": Object {
        "east": -122.28178000000003,
        "north": 37.9298239,
        "south": 37.6398299,
        "west": -123.17382499999997,
      },
      "location": Object {
        "lat": 37.7749295,
        "lng": -122.41941550000001,
      },
      "location_type": "APPROXIMATE",
      "viewport": Object {
        "east": -122.34820000000002,
        "north": 37.812,
        "south": 37.70339999999999,
        "west": -122.52699999999999,
      },
    },
    "place_id": "ChIJIQBpAG2ahYAR_6128GcTUEo",
    "types": Array [
      "locality",
      "political",
    ],
  },
]
`;
